### PR TITLE
Add ignore errors parameter

### PR DESCRIPTION
**Pull request by DocRoms**
We want to add an `ignore_cache_error` parameter for deactivate Exception from redis when the cache is empty.

## Todo ##
* [x] Fix composer PHP version 
* [x] Add `ignore_cache_errors` parameter in configuration 
* [x] Use this new parameter on the bundle
* [x] Add event for monitor this functionality.
* [x] Fix and edit tests

### CODE REVIEW

File: src/DependencyInjection/M6WebGuzzleHttpExtension.php [hunk 1]
```diff
@@ -158,19 +158,21 @@ protected function setGuzzleProxyHandler(ContainerBuilder $container, $clientId,
...
                 ));
             }
 
-            $curlhandler->addMethodCall('setCache', [$cacheService, $defaultTtl, $headerTtl, $cacheServerErrors, $cacheClientErrors]);
-            $curlMultihandler->addMethodCall('setCache', [$cacheService, $defaultTtl, $headerTtl, $cacheServerErrors, $cacheClientErrors]);
+            $defaultTtl = $config['guzzlehttp_cache']['default_ttl'];
+            $headerTtl = $config['guzzlehttp_cache']['use_header_ttl'];
+            $cacheServerErrors = $config['guzzlehttp_cache']['cache_server_errors'];
+            $cacheClientErrors = $config['guzzlehttp_cache']['cache_client_errors'];
+            $ignoreCacheErrors = $config['guzzlehttp_cache']['ignore_cache_errors'] ?? false;
```
Comment (fabdsp) [inline src/DependencyInjection/M6WebGuzzleHttpExtension.php]: Why "?? : false" - The default value should assure there is always true or false, no ?

File: src/Handler/CurlHandler.php [hunk 2]
```diff
@@ -2,11 +2,27 @@
...
+        parent::__construct($options);
+    }
+
     use CacheTrait;
+
+    public function getEventDispatcher(): EventDispatcherInterface
+    {
+        return $this->eventDispatcher;
+    }
 }
```
Comment (f-dumas) [inline src/Handler/CurlHandler.php]: :scream:

File: src/Handler/CurlMultiHandler.php [hunk 3]
```diff
@@ -2,11 +2,27 @@
...
+        parent::__construct($options);
+    }
+
     use CacheTrait;
+
+    public function getEventDispatcher(): EventDispatcherInterface
+    {
+        return $this->eventDispatcher;
+    }
 }
```
Comment (f-dumas) [inline src/Handler/CurlMultiHandler.php]: :scream_cat:

File: README.md [hunk 4]
```diff
@@ -179,11 +179,11 @@ m6web_guzzlehttp:
         default:
             base_uri: ""                           # Base uri to prepend on request uri
             timeout: 5.0                           # request timeout
-            http_errors: true                      # enable or disable exception on http errors
+            http_errors: true                      # set to false to disable throwing exceptions on an HTTP protocol errors
```
Comment (pabedu) [inline README.md]: remove an :)

File: README.md [hunk 5]
```diff
@@ -179,11 +179,11 @@ m6web_guzzlehttp:
...
             redirect_handler: curl                 # guzzle or curl
-            proxy: proxy:port                      # Optional. Set the proxy for client.
             guzzlehttp_cache:                      # optional cache
                 service: '@my_cache_service'       # reference to service who implements the cache interface
+                ignore_cache_errors: false          # at true, no Exception has been thrown when cache is unavailable
```
Comment (pabedu) [inline README.md]: would be throw*

File: composer.json [hunk 6]
```diff
@@ -12,7 +12,7 @@
     }
   ],
   "require": {
-    "php": ">=5.5",
+    "php": ">=7.0",
```
Comment (pabedu) [inline composer.json]: that's a public bundle, is there a need for it ? some people might be using it with older version of php, we don't want to be unkind to those people :)

File: src/EventDispatcher/GuzzleCacheEvent.php [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+    {
+        $this->request = $request;
+    }
+
+    /**
+     * Get Request
+     *
+     * @return RequestInterface
+     */
+    public function getRequest() : RequestInterface
```
Comment (pabedu) [inline src/EventDispatcher/GuzzleCacheEvent.php]: 😱

File: src/Handler/CacheTrait.php [hunk 8]
```diff
@@ -226,8 +215,19 @@ public function __invoke(RequestInterface $request, array $options)
...
-        if ($response = $this->getCached($request)) {
-            return new FulfilledPromise($response);
+        try {
+            if ($response = $this->getCached($request)) {
+                return new FulfilledPromise($response);
+            }
+        }catch(\Exception $e) {
+            if (!$this->isIgnoreCacheErrors){
+                throw $e;
+            }else{
```
Comment (pabedu) [inline src/Handler/CacheTrait.php]: not beautiful ;)

File: src/Handler/CurlMultiHandler.php [hunk 9]
```diff
@@ -2,11 +2,27 @@
...
+        $this->eventDispatcher = $eventDispatcher;
+
+        parent::__construct($options);
+    }
+
     use CacheTrait;
-}
\ No newline at end of file
+
+    public function getEventDispatcher(): EventDispatcherInterface
```
Comment (pabedu) [inline src/Handler/CurlMultiHandler.php]: trait should be above contruct or event atributes I think, same in CurlHandler.php

File: composer.json (hunk 6 above)
Comment (Meroje) [inline composer.json]: Should have been tagged as major 😢

Comment (pabedu) [reply composer.json]: a few tags before indeed :(

**Pull request merged**